index.js

```diff
@@ -9,20 +9,60 @@ let validator = require('validator');
 const TAB = '    ';
 const helpRegex = /^\[(string|number|boolean|json)\].*$/i;
 
-function recurse(obj, counter, path) {
+function appendExtraData(tmpl, extra) {
+	var returnObject = {};
+	if (!extra) extra = {};
+	for (let key in tmpl) {
+		if (tmpl.hasOwnProperty(key)) {
+			if (_.isObject(tmpl[key])) {
+				returnObject[key] = appendExtraData(tmpl[key], extra[key]);
+			} else {
+				returnObject[key] = tmpl[key];
+			}
+		}
+	}
+	for (let key in extra) {
+		if (extra.hasOwnProperty(key)) {
+			if (_.isObject(extra[key])) {
+				if (!tmpl[key]) tmpl[key] = {};
+				if (!tmpl[key].toString().match(/^\[json\].*/))
+					returnObject[key] = appendExtraData(tmpl[key], extra[key]);
+			} else {
+				if (!tmpl.hasOwnProperty(key)) returnObject[key] = extra[key];
+			}
+		}
+	}
+	return returnObject;
+}
+
+function setDefaultValues(parsedTmpl, values) {
```

Comment:
Can we re-write this function like this?
```js
function setDefaultValues(parsedTmpl, values) {
	for (let i=0; i<parsedTmpl.lines.length; i++) {
		const line = parsedTmpl.lines[i];
		let obj = values;
		for(let j=0; j<line.path.length; j++) {
			obj = obj[line.path[j]];
		}
		line.value = obj.toString();
	}
}
```

It's a little bit more code but I think it's more clear what's going on than with multiple variables changing inside each for loop.